docs/types/buffer.md

```diff
@@ -6,9 +6,7 @@ There are two types of buffers: `memory` buffers and `disk` buffers.
 
 ## Memory Buffers
 
-Memory buffers keep log entries in memory until they are flushed, which makes them very fast. However, because
-entries are only stored in memory, they will be lost if the agent is shut down uncleanly. If the agent is shut down
-cleanly, they will be saved to the agent's database.
+Memory buffers keep log entries in memory, which makes them very fast. Upon a clean shutdown the buffer will dump any remaining entries to the operator using it for the operator to finish processing.
```

Comment:
```suggestion
Memory buffers keep log entries in memory, which makes accessing them very fast. Upon a clean shutdown the buffer will dump any remaining entries to the operator using it for the operator to finish processing.
```
```suggestion
Memory buffers store log entries in memory, which makes them very fast. On a clean shutdown, the buffer will drain any remaining entries the operator is buffering so that the operator finishes processing them.
```

I believe this is still the behavior correct? I was mostly a tad bit confused on the wording of the last part with a double reference to operator. Let me know if you think this is an appropriate re-wording!